### ISSUE TITLE
Running tests with `cargo-nextest`

### ISSUE THREAD

**Issue by NobodyXu** [xd009642/tarpaulin#1280]
`cargo-nextest` can dramatically speedup testing, especially for projects with a lot of tests, though it currently doesn't support running doc-tests in parallel.

Is it possible for `cargo-tarpaulin` to use `cargo-nextest` to run tests in parallel and then run `cargo test --doc ...` for doc-tests?

Or maybe `cargo-tarpaulin` can reuse infrastructure created by `cargo-nextest` to speedup itself?

**Comment by xd009642** [comment]
I think this is a duplicate (there should be another one somewhere), I've slightly put off doing this because it would be implemented only for the llvm coverage and not ptrace based coverage (nextest won't work well with ptrace based tracers like kcov or tarpaulins ptrace engine). Just means another set of mutually exclusive configuration that needs to be figured out

**Comment by NobodyXu** [comment]
> not ptrace based coverage (nextest won't work well with ptrace based tracers like kcov or tarpaulins ptrace engine). Just means another set of mutually exclusive configuration that needs to be figured out

Perhaps you can use [nextest-runner](https://docs.rs/nextest-runner) to implement this directly in this crate?

**Comment by NobodyXu** [comment]
> hmm maybe, I'll have to look into it - wonder how the Noop signal handler affects the usage of the rest of nextest 🤔

Not sure, but AFAIK [`nextest-runner` uses `tokio::signal`](https://github.com/search?q=repo%3Anextest-rs%2Fnextest%20tokio%3A%3Asignal&type=code).

So if you also use `tokio::signal`, it should be ok.

**Comment by xd009642** [comment]
I do not, for it to work with ptrace based traces nextest needs to ignore all signals otherwise tarpaulin can't handle them - tarpaulin's just using ptrace directly

**Comment by NobodyXu** [comment]
@xd009642 [`nextest-runner` does support registering no-op signal handler](https://github.com/nextest-rs/nextest/blob/c54694dfe7be016993983b5dedbcf2b50d4b1a6e/nextest-runner/src/signal.rs#L19).

`tokio::process` would still register a signal handler for `SIGCHLD` and call `waitpid`, that's indeed a problem.

Perhaps you can get in contact with upstream and see if they are willing to support this use case?

**Comment by RobbieMcKinstry** [comment]
https://github.com/xd009642/tarpaulin/issues/992 is the issue that this is a duplicate of. 👍🏻  ❤️

**Comment by NobodyXu** [comment]
> #992 is the issue that this is a duplicate of. 👍🏻 ❤️

Thanks, I will close this as duplicate
